src/net_processing.cpp

```diff
@@ -396,6 +397,10 @@ limitedmap<uint256, std::chrono::microseconds> g_already_asked_for GUARDED_BY(cs
 /** Map maintaining per-node state. */
 static std::map<NodeId, CNodeState> mapNodeState GUARDED_BY(cs_main);
 
+/** In-memory cache of all BIP157 compact filter checkpoints for the active chain. */
+static std::vector<std::pair<const CBlockIndex*, uint256>> active_chain_cf_headers;
```

Comment:
8788254

What the perf improvement of caching checkpoints ? If clients are configured with trusted checkpoints, they are only going to use the higher range, or maybe a rolling checkpoints for last X headers/filters would be better as they are likely to be requested by everyone

Reply:
The `GETCFCHECKPT` request returns all filter checkpoints. There's no parameter to request a subset from a different start height because the message is small and only requested once per catchup generally.